src/tests/mod.rs

```diff
@@ -1,27 +1,47 @@
 use crate::compiler::{AmberCompiler, CompilerOptions};
 extern crate test_generator;
+use heraclitus_compiler::prelude::Message;
 use itertools::Itertools;
 use std::fs;
 use std::path::PathBuf;
 use std::process::{Command, Stdio};
 
 pub mod cli;
-pub mod errors;
 pub mod extra;
 pub mod postprocessor;
 mod stdlib;
 mod validity;
+mod erroring;
 
-/// compare the output of the given code with the expected output
-pub fn test_amber(code: impl Into<String>, result: impl AsRef<str>) {
+pub enum TestOutcomeTarget {
+    Success,
+    Failure,
+}
+
+pub fn eval_amber_code(code: &str) -> Result<String, Message> {
     let options = CompilerOptions::default();
-    let mut compiler = AmberCompiler::new(code.into(), None, options);
-    match compiler.test_eval() {
-        Ok(eval_result) => assert_eq!(
-            eval_result.trim_end_matches('\n'),
-            result.as_ref().trim_end_matches('\n'),
-        ),
-        Err(err) => panic!("ERROR: {}", err.message.unwrap()),
+    let mut compiler = AmberCompiler::new(code.to_string(), None, options);
+    compiler.test_eval()
+}
+
+/// Tests script output in case of success or failure
+pub fn test_amber(code: &str, result: &str, target: TestOutcomeTarget) {
```

Comment:
You can reduce the amount of indentation by calling `eval_amber_code()` at the top of the function, and collapsing the `match` statements.  I would also suggest using consistent braces, so the results (successful or `panic!`ing) are all at the same level:
```
/// Tests script output in case of success or failure
pub fn test_amber(code: &str, result: &str, target: TestOutcomeTarget) {
    let result = eval_amber_code(code);
    match target {
        TestOutcomeTarget::Success => match result {
            Ok(eval_result) => assert_eq!(
                eval_result.trim_end_matches('\n'),
                result.trim_end_matches('\n'),
            )
            Err(err) => {
                panic!("ERROR: {}", err.message.unwrap())
            }
        }
        TestOutcomeTarget::Failure => match result {
            Ok(eval_result) => {
                panic!("Expected error, got: {}", eval_result)
            }
            Err(err) => {
                assert_eq!(err.message.expect("Error message expected"), result)
            }
        }
    }
}
```